### ISSUE TITLE
cluster IP configured on the kube-dummy-if should not respond to ARP requests

### ISSUE THREAD

**Issue by murali-reddy** [cloudnativelabs/kube-router#561]
cluster IP configured on the kube-dummy-if should not respond to ARP requests.

`kube-dummy-if` interface is configured to be `NOARP` but that is not sufficient condition

```
504: kube-dummy-if: <BROADCAST,NOARP,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN mode DEFAULT group default 
    link/ether 7a:2d:83:f0:c8:f0 brd ff:ff:ff:ff:ff:ff
```
see the conversation https://kubernetes.slack.com/archives/C8DCQGTSB/p1535088493000100

**Comment by asteven** [comment]
We run some keepalived based loadbalancers in DSR mode.
I have to configure my real-servers like this to prevent them for ARP'ing around:

```
net.ipv4.conf.all.arp_ignore=1
net.ipv4.conf.all.arp_announce=2
```

I assume the kube-dummy-if would need something like this to?
e.g.

```
net.ipv4.conf.kube-dummy-if.arp_ignore = 1
net.ipv4.conf.kube-dummy-if.arp_announce = 2
```

Not sure if this would have side effects though. I do not yet have the full picture how/where/in what context kube-dummy-if is used.

**Comment by asteven** [comment]
Some context:
http://kb.linuxvirtualserver.org/wiki/Using_arp_announce/arp_ignore_to_disable_ARP

**Comment by murali-reddy** [comment]
thanks @kvaps for pointing out the PR

**Comment by roffe** [comment]
solved by #580
